File: fritzStreams/src/main/java/de/acepe/fritzstreams/backend/json/OnDemandStreamDescriptor.java
package de.acepe.fritzstreams.backend.json;

import java.util.List;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class OnDemandStreamDescriptor {

    @SerializedName("_sortierArray")
    @Expose
    private List<Integer> sortierArray = null;
    @SerializedName("_defaultQuality")
    @Expose
    private List<String> defaultQuality = null;
    @SerializedName("rbbtitle")
    @Expose
    private String rbbtitle;
    @SerializedName("rbbhandle")
    @Expose
    private String rbbhandle;
    @SerializedName("_type")
    @Expose
    private String type;
    @SerializedName("_isLive")
    @Expose
    private Boolean isLive;
    @SerializedName("_dvrEnabled")
    @Expose
    private Boolean dvrEnabled;
    @SerializedName("_geoblocked")
    @Expose
    private Boolean geoblocked;
    @SerializedName("_duration")
    @Expose
    private Integer duration;
    @SerializedName("_audioImage")
    @Expose
    private String audioImage;
    @SerializedName("_mediaArray")
    @Expose
    private List<MediaArray> mediaArray = null;

    private List<Integer> getSortierArray() {
        return sortierArray;
    }

    public void setSortierArray(List<Integer> sortierArray) {
        this.sortierArray = sortierArray;
    }

    public List<String> getDefaultQuality() {
        return defaultQuality;
    }

    public void setDefaultQuality(List<String> defaultQuality) {
        this.defaultQuality = defaultQuality;
    }

    public String getRbbtitle() {
        return rbbtitle;
    }

    public void setRbbtitle(String rbbtitle) {
        this.rbbtitle = rbbtitle;
    }

    public String getRbbhandle() {
        return rbbhandle;
    }

    public void setRbbhandle(String rbbhandle) {
        this.rbbhandle = rbbhandle;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public Boolean getLive() {
        return isLive;
    }

    public void setLive(Boolean live) {
        isLive = live;
    }

    public Boolean getDvrEnabled() {
        return dvrEnabled;
    }

    public void setDvrEnabled(Boolean dvrEnabled) {
        this.dvrEnabled = dvrEnabled;
    }

    public Boolean getGeoblocked() {
        return geoblocked;
    }

    public void setGeoblocked(Boolean geoblocked) {
        this.geoblocked = geoblocked;
    }

    public Integer getDuration() {
        return duration;
    }

    public void setDuration(Integer duration) {
        this.duration = duration;
    }

    public String getAudioImage() {
        return audioImage;
    }

    public void setAudioImage(String audioImage) {
        this.audioImage = audioImage;
    }

    public List<MediaArray> getMediaArray() {
        return mediaArray;
    }

    public void setMediaArray(List<MediaArray> mediaArray) {
        this.mediaArray = mediaArray;
    }
}

File: fritzStreams/src/main/java/de/acepe/fritzstreams/backend/json/OnDemandStreamDescriptor.kt
package de.acepe.fritzstreams.backend.json

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class OnDemandStreamDescriptor {

    @SerializedName("_sortierArray")
    @Expose
    private var sortierArray: List<Int>? = null
        set
    @SerializedName("_defaultQuality")
    @Expose
    var defaultQuality: List<String>? = null
    @SerializedName("rbbtitle")
    @Expose
    var rbbtitle: String? = null
    @SerializedName("rbbhandle")
    @Expose
    var rbbhandle: String? = null
    @SerializedName("_type")
    @Expose
    var type: String? = null
    @SerializedName("_isLive")
    @Expose
    var live: Boolean? = null
    @SerializedName("_dvrEnabled")
    @Expose
    var dvrEnabled: Boolean? = null
    @SerializedName("_geoblocked")
    @Expose
    var geoblocked: Boolean? = null
    @SerializedName("_duration")
    @Expose
    var duration: Int? = null
    @SerializedName("_audioImage")
    @Expose
    var audioImage: String? = null
    @SerializedName("_mediaArray")
    @Expose
    var mediaArray: List<MediaArray>? = null
}